plugins/GigPlayer/GigPlayer.cpp

```diff
@@ -526,107 +509,103 @@ void GigInstrument::loadSample( GigSample& sample, SampleFrame* sampleData, f_cn
 		}
 	}
 
-	unsigned long allocationsize = samples * sample.sample->FrameSize;
-	int8_t buffer[allocationsize];
+	const auto allocationSize = samples * sample.sample->FrameSize;
+	const auto neededCapacity = static_cast<std::vector<int8_t>::size_type>(allocationSize);
```

Comment:
`allocationSize` is already `std::size_t` so it looks like there's no need for the static cast. And the `neededCapacity` variable may be unnecessary if you can just use `allocationSize`.

Reply:
I would defer this point to @Veratil whenever he returns as I don't know about the inner workings.